string barcode_node
string goal_node
string current_node
